src/toil/test/utils/utilsTest.py

```diff
@@ -183,6 +186,128 @@ def testAWSProvisionerUtils(self):
+        self.assertEqual(ToilStatus.getStatus(jobstoreLoc), 'RUNNING')
+        wf.wait()
+        self.assertEqual(ToilStatus.getStatus(jobstoreLoc), 'ERROR')
+        shutil.rmtree(jobstoreLoc)
+
+    @needs_cwl
+    @needs_docker
+    def testGetStatusFailedCWLWF(self):
+        """
+        Test that ToilStatus.getStatus() behaves as expected with a failing CWL workflow.
+
+        Run a cwl workflow, with the badWorker option set to force a failure, in a subprocess. The workflow is allowed
+        to continue for 2 seconds to allow its jobstore to  be created. Pausing the process at this point allows
+        getStatus() to check the jobstore without the workflow  completing. A 'RUNNING' responses is expected. The
+        process is resumed, allowed to continue to completion and, 'ERROR' is expected in return.
+        """
+        files = ['src/toil/test/cwl/sorttool.cwl', 'src/toil/test/cwl/whale.txt']
+        jobstoreName = 'failing-cwl-js'
+        jobstoreLoc = os.path.join(os.getcwd(), jobstoreName)
+        cmd = ['toil-cwl-runner', '--jobStore', jobstoreLoc, '--clean', 'never', '--badWorker', '1', files[0],
+               '--reverse', '--input', files[1]]
+        wf = Popen(cmd)
+        wfRun = psutil.Process(pid=wf.pid)
+        time.sleep(2)
+        wfRun.suspend()
+        self.assertEqual(ToilStatus.getStatus(jobstoreLoc), 'RUNNING')
+        wfRun.resume()
+        time.sleep(10)
+        self.assertEqual(ToilStatus.getStatus(jobstoreLoc), 'ERROR')
+        shutil.rmtree(jobstoreLoc)
+
+    def testGetStatusSuccessfulToilWF(self):
+        """
+        Test that ToilStatus.getStatus() behaves as expected with a successful Toil workflow.
+
+        Run a Toil workflow in a subprocess. While this workflow  could be called by importing and evoking its main
+        function, doing so would remove the opprotunity to test the 'RUNNING' functionality of getStatus(). The workflow
+        is allowed to continue for 2 seconds to allow its jobstore to  be created. A 'RUNNING' responses is expected.
+        The process is allowed to continue to completion and, 'ERROR' is expected in return.
+        """
+        jobstoreName = 'successful-toil-js'
+        jobstoreLoc = os.path.join(os.getcwd(), jobstoreName)
+        cmd = ['python', '-m', 'toil.test.sort.sort', 'file:' + jobstoreName, '--clean', 'never']
+        wf = Popen(cmd)
+        time.sleep(2)  # Need to let jobstore be created before checking its contents.
+        self.assertEqual(ToilStatus.getStatus(jobstoreLoc), 'RUNNING')
+        wf.wait()
+        self.assertEqual(ToilStatus.getStatus(jobstoreLoc), 'COMPLETED')
+        shutil.rmtree(jobstoreLoc)
+
+    @needs_cwl
+    @needs_docker
+    def testGetStatusSuccessfulCWLWF(self):
+        """
+        Test that ToilStatus.getStatus() behaves as expected with a successful CWL workflow.
+
+        Run a cwl workflow in a subprocess. The workflow is allowed to continue for 2 seconds to allow its jobstore to
```

Comment:
Line 289 is good, but the documentation from L291-294 should be shown as inline comments.